### ISSUE TITLE
Pylint ignores disable=invalid-name under certain conditions 

### ISSUE THREAD

**Issue by ya3xp** [pylint-dev/pylint#9861]
### Bug description

```python
Consider following structure:


|_ mymodule
   |_ __init__.py
   |_ example.py
   |_ myclasses.py
```

`__init__.py` is empty

`myclasses.py` :

class ClassA:
    badClassVariable = 0

```


`example.py` :
```python
from mymodule.myclasses import ClassA

class ClassB:
    badClassVariable = 0

example_a = ClassA()
example_a.badClassVariable = 1  # pylint: disable=invalid-name

example_b = ClassB()
example_b.badClassVariable = 1  # pylint: disable=invalid-name

badVariable = 1  # pylint: disable=invalid-name

```

Pylint will flag `example_a.badClassVariable` on line 7 despite `disable=invalid-name`.

Pylint will honor `disable=invalid-name` on lines 10 and 12 and will not flag `example_b.badClassVariable` nor `badVariable`.
```


### Configuration

_No response_

### Command used

```shell
pylint --disable=all --enable=C0103 mymodule
```

### Pylint output

```python
************* Module mymodule.example
mymodule/example.py:7:0: C0103: Attribute name "badClassVariable" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

Pylint should not report an issue for line 7, as it already does for lines 10 and 12.

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)]
pylintbug-py3.12
```

### OS / Environment

Issue occurred on both MacOS 14.6.1 and containereized Alpine linux 3.18.

### Additional dependencies

```python
dill==0.3.8
isort==5.13.2
mccabe==0.7.0
platformdirs==4.2.2
tomlkit==0.13.0
```

**Comment by emin63** [comment]
Any update on this? I think I hit a similar issue. The only thing that I could get to work was to disable `invalid-name` in my `.pylintrc`. Thanks.
